src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy

```diff
@@ -306,28 +312,52 @@ class IntelliJPlugin implements Plugin<Project> {
-        configurePrepareSandboxTask(project, extension, true)
+        configurePrepareSandboxTask(project, extension, PREPARE_SANDBOX_TASK_NAME, "")
+        configurePrepareSandboxTask(project, extension, PREPARE_TESTING_SANDBOX_TASK_NAME, "-test")
+        addRobotServerDependency(project, "0.9.1")
+        configurePrepareSandboxTask(project, extension, PREPARE_UI_TESTING_SANDBOX_TASK_NAME, "-uiTest")
+    }
+
+    private static void addRobotServerDependency(@NotNull Project project, @NotNull String robotServerVersion) {
+        Utils.info(project, "Configuring robot-server dependency")
+
+        project.repositories {
+            maven { url "https://jetbrains.bintray.com/intellij-third-party-dependencies" }
+        }
+
+        project.configurations {
+            robotServerPluginImplementation
+        }
+
+        project.dependencies {
+            robotServerPluginImplementation("org.jetbrains.test:robot-server-plugin:$robotServerVersion")
+        }
     }
 
     private static void configurePrepareSandboxTask(@NotNull Project project,
                                                     @NotNull IntelliJPluginExtension extension,
-                                                    boolean inTest) {
+                                                    String taskName,
+                                                    String testSuffix) {
         Utils.info(project, "Configuring prepare sandbox task")
-        def taskName = inTest ? PREPARE_TESTING_SANDBOX_TASK_NAME : PREPARE_SANDBOX_TASK_NAME
-        project.tasks.create(taskName, PrepareSandboxTask).with {
+        project.tasks.create(taskName, PrepareSandboxTask).with { task ->
             group = GROUP_NAME
             description = "Prepare sandbox directory with installed plugin and its dependencies."
             conventionMapping('pluginName', { extension.pluginName })
             conventionMapping('pluginJar', { (project.tasks.findByName(JavaPlugin.JAR_TASK_NAME) as Jar).archivePath })
-            conventionMapping('destinationDir', { project.file(Utils.pluginsDir(extension.sandboxDirectory, inTest)) })
-            conventionMapping('configDirectory', { Utils.configDir(extension.sandboxDirectory, inTest).toString() })
+            conventionMapping('destinationDir', { project.file(Utils.pluginsDir(extension.sandboxDirectory, testSuffix)) })
+            conventionMapping('configDirectory', { Utils.configDir(extension.sandboxDirectory, testSuffix).toString() })
             conventionMapping('librariesToIgnore', { project.files(extension.ideaDependency.jarFiles) })
             conventionMapping('pluginDependencies', { extension.pluginDependencies })
             dependsOn(JavaPlugin.JAR_TASK_NAME)
+
+            if (taskName == PREPARE_UI_TESTING_SANDBOX_TASK_NAME) {
+                from project.zipTree(project.configurations.robotServerPluginImplementation.files.find { it.name.endsWith("zip") })
```

Comment:
there is a `singleFile` method, it might fit

Reply:
copying is also should be delayed to actually run. see `pluginDependencies` convention mapping, you should fill it with your plugin